apps/email/js/cards/lst/msg_click.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+define(function(require, exports) {
+
+var cards = require('cards'),
+    ConfirmDialog = require('confirm_dialog'),
+    largeMsgConfirmMsgNode = require('tmpl!../msg/large_message_confirm.html');
+
+/**
+ * Show a warning that the given message is large.
+ * Callback is called with cb(true|false) to continue.
+ */
+function showLargeMessageWarning(size, cb) {
+  var dialog = largeMsgConfirmMsgNode.cloneNode(true);
+  // TODO: If UX designers want the size included in the warning
+  // message, add it here.
+  ConfirmDialog.show(dialog,
+    { // Confirm
+      id: 'msg-large-message-ok',
+      handler: function() { cb(true); }
+    },
+    { // Cancel
+      id: 'msg-large-message-cancel',
+      handler: function() { cb(false); }
+    }
+  );
+}
+
+/**
+ * A mixin for lst cards that want tapping on a message to go to a message
+ * reader. Check all the `this` references below to see implicit dependencies
+ * on instance state.
+ * A longer range TODO is to remove or make explicit the dependencies.
+ */
+  return {
+    onClickMessage: function(messageNode, event) {
+      // You cannot open a message if this is the outbox and it is syncing.
+      if (this.curFolder &&
+          this.curFolder.type === 'outbox' && this.outboxSyncInProgress) {
+        return;
+      }
+
+      var header = messageNode.message;
+
+      // Skip nodes that are default/placeholder ones.
+      if (header && header.isPlaceholderData) {
+        return;
+      }
+
+      if (this.editMode) {
+        var idx = this.selectedMessages.indexOf(header);
```

Comment:
I'm honestly not sure if it will make things appreciably cleaner, but I think it would be better to move the logic inside the conditional here up into the edit controller since that's where all these properties are defined and otherwise manipulated.